lib/realtime/api/presence.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Realtime.Api.Presence do
+  @moduledoc """
+  Defines the Presence schema
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  @derive {Jason.Encoder, only: [:inserted_at, :updated_at, :id, :channel_id]}
+
+  @type t :: %__MODULE__{}
+
+  @schema_prefix "realtime"
+  schema "presences" do
+    field(:check, :boolean, default: false)
+    timestamps()
+
+    belongs_to(:channel, Realtime.Api.Channel)
+  end
+
+  def changeset(broadcast, attrs) do
+    broadcast
+    |> cast(attrs, [:check, :inserted_at, :updated_at, :channel_id])
```

Comment:
This is a bit out of the scope of the PR (*) but perhaps worth taking a step back and talking about casting. Ecto has a nice [External vs internal data](https://hexdocs.pm/ecto/Ecto.Changeset.html#module-external-vs-internal-data) section and long story short is: in our experience it's useful to have a clear distinction between internal and external data and use _casting_ exclusively for the latter. When we work with internal data by definition it already has proper types and values within allowed sets, if it ain't, it's a programming bug. In other words, if we somehow got an invalid value it shouldn't be gracefully handled (which casting does), it's a programming bug that we should fix.

Btw, is there a use case for changing inserted_at/updated_at by callers? They tend to be automatically handled by Ecto. And if you really need to do it on a special occasion, you can always do `change(struct, inserted_at: ...)`

Aside from internal vs external data, there's also the matter of casting FKs like channel_id. In this particular PRs it's fine but we believe it's a good habit to build to almost never do that (and when you do, explain in a comment) because it might open a security hole. A malicious user could pass a `channel_id` that belongs to another tenant and gain access. (This might be totally moot given RLS but I believe it might be instructive anyway.) A quick fix is:

```diff
- Broadcast.changeset(%Broadcast{}, %{channel_id: channel.id})
+ Broadcast.changeset(%Broadcast{channel_id: channel.id}, %{})
```

and conceptually it is: `channel.id` is a value from struct (internal data), `%Broadcast{}` is a struct (internal data). In this case we don't have any external data so we might as well do:

```elixir
change(%Broadcast{channel_id: channel.id})
```

(*) We have similar structure in Api.Broadcast and so it's worth it to keep it uniform. Also apologies if we already had similar discussions in the past!

Reply:
This was really informative! Thank you!

Probably will do it in a different PR but indeed this changes sound important so will apply them to the schemas that need them